### PR TITLE
Nc  bump golang

### PR DESCRIPTION
**Pull request by SocalNick**

### CODE REVIEW

**Comment by gmacdougall** [comment]
I think that the VERSION in `committer.go` still needs to be updated. On the plus side, this builds for me:

```
❯ go build ./committer.go
❯ ./committer --version
0.1.7
```

**Review by gmacdougall — APPROVED** [review]
It builds and it runs which is a big step from before this PR. Thanks Nicholas!

:+1:

File: .github/workflows/publish.yml [hunk 1]
```diff
@@ -28,13 +28,11 @@ jobs:
 
       - name: Install go
         run: |
-          curl -L -o /tmp/go.pkg https://golang.org/dl/go1.16.darwin-amd64.pkg
+          curl -L -o /tmp/go.pkg https://golang.org/dl/go1.17.darwin-amd64.pkg
```
Comment (williamtheaker) [inline .github/workflows/publish.yml:31]: The [macOS](https://github.com/actions/virtual-environments/blob/main/images/macos/macos-10.15-Readme.md) and [Ubuntu](https://github.com/actions/virtual-environments/blob/main/images/linux/Ubuntu2004-README.md) runners have Go 1.17.3 cached. If none of the dependencies require building on Darwin, you could switch over to Ubuntu runners at 10% of the price.

Comment (SocalNick) [reply .github/workflows/publish.yml:31]: Hey @williamtheaker - I didn't set this up, but I think the `build.sh` [step depends on a tool called `lipo` to make a universal macOS binary](https://github.com/Gusto/committer/blob/master/.github/build.sh#L12) (arm64 + amd64). This tool is from Xcode, so I imagine it's macOS only.

**Review by williamtheaker — APPROVED** [review]

**Pull request merged**